src/Microsoft.DotNet.Build.Tasks/GetTargetMachineInfo.cs

```diff
@@ -29,6 +48,18 @@ public override bool Execute()
             else if (RuntimeInformation.IsOSPlatform(OSPlatform.Create("NETBSD")))
                 TargetOS = "NetBSD";
 
+            if (TargetArch == null)
+            {
+                Log.LogError("{0} is null", nameof(TargetArch));
```

Comment:
Minor but when updating this in the future we should make these messages more useful. At least say why it is null otherwise folks will still not know why. I'm thinking something simple like "TargetArch/OS is unknown"